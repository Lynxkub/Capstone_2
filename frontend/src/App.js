import React , {useState , useEffect} from 'react';
import './App.css';
import Routes from './Routes';
import NavBar from './NavBar';
import FoodlyApi from './api';


function App() {

  const [userToken , setUserToken] = useState(localStorage.getItem('token'));
  const [username , setUsername] = useState(localStorage.getItem('username'));


  useEffect(() => {
    localStorage.setItem('token' , userToken)
  } , [userToken])


  FoodlyApi.token = localStorage.getItem('token');



  const signUp = async (newUser) => {
    const generatedToken = await FoodlyApi.register(newUser);
    setUserToken(generatedToken);
    localStorage.setItem('token' , generatedToken);
    setUsername(localStorage.getItem('username'));
  }

  const login = async (returningUser) => {
    const generatedToken = await FoodlyApi.authenticate(returningUser);
    setUserToken(generatedToken)
    localStorage.setItem('token' , FoodlyApi.token);
    setUsername(localStorage.getItem('username'));
  }


  const logout = () => {
    setUserToken(undefined);
    setUsername(localStorage.getItem('username'));
  }

  const addComment = async (comment) => {
    const createComment = await FoodlyApi.makeComment(comment);
    return createComment;
  }

  

  
  return (
    <div className="fill-window">
      <NavBar />
    <Routes signUp = {signUp} token = {userToken} login = {login}  logout = {logout} addComment = {addComment}/>
    </div>
  );
}

export default App;