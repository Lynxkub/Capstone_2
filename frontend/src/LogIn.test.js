import React from 'react';
import {render , fireEvent} from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect';
import LogInForm from './LoginForm';
import App from './App';
import FoodlyApi from './api';



test('it renders without crashing' , () => {
    render(
        <MemoryRouter>
            <LogInForm />
        </MemoryRouter>
    )
})



test('it matches snapshot' , () => {
    const { asFragment } = render (
        <MemoryRouter>
           <LogInForm />
        </MemoryRouter>
    )

    expect(asFragment()).toMatchSnapshot();
})



test('it logs user in successfully' , () => {
    const {getByText , getByTestId} = render (
        <MemoryRouter initialEntries={['/']}>
            <App />
        </MemoryRouter>
    )


    const loginLink = getByText('Log In');
    fireEvent.click(loginLink);

    // ensure that we are on the log in form

    const username = getByTestId('username');
    expect(username).toBeInTheDocument();

        // fill out form and submit

    const password = getByTestId('password');
    const loginButton =  getByTestId('loginSubmit')

    fireEvent.change(username , {target : {value : 'test'}})
    fireEvent.change(password , {target : {value : 'testtest'}})
    fireEvent.click(loginButton);

    // check to make sure that we are redirected back to the home page

    setTimeout(() => {
        const newHomePage = getByText('Welcome back Test!')
        expect(newHomePage).toBeInTheDocument();
    } , 1000)
 
})