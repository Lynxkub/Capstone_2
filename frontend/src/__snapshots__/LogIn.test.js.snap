// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="username"
      >
        Username
      </label>
      <input
        data-testid="username"
        name="username"
        placeholder="Username"
        type="text"
        value=""
      />
      <label
        for="password"
      >
        Password
      </label>
      <input
        data-testid="password"
        name="password"
        placeholder="Password"
        type="password"
        value=""
      />
      <button
        data-testid="loginSubmit"
      >
        Log In
      </button>
    </form>
  </div>
</DocumentFragment>
`;
